### PR TITLE
DP based C program to find minimum operations required to convert one string to other string.

### PR DESCRIPTION
**Pull request by nutanaarohi123**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
YES

### Description
#439 
 A Dynamic Programming based C program to find a minimum number of operations to convert string1 to string2.
**Time Complexity** :  O(mn)
**Space Complexity**:  O(mn)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#439

### CODE REVIEW

File: C/dp/EditDistanceDP.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
+/* A Dynamic Programming based C program to find minimum
```
Comment (iamrajiv) [inline C/dp/EditDistanceDP.c]: @nutanaarohi123 format like this 
```
/* 
A Dynamic Programming based C program to find the minimum 
number operations to convert string1 to string2 
*/
```
Also, break line after this

File: C/dp/EditDistanceDP.c [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        printf("\n Enter the first string:");
+        scanf("%[^\n]%*c", string1);
+        printf("\n Enter the second string:");
+        scanf("%[^\n]%*c", string2);
+        n=editDistDP(string1, string2, strlen(string1), strlen(string2)); 
+        printf("\n Output:");
+        printf(" %d\n ", n);
+	return 0; 
+} 
+/* Input:
```
Comment (iamrajiv) [inline C/dp/EditDistanceDP.c]: @nutanaarohi123 format like this
```
/*
 Input:
Enter the first string: "Sunday"
Enter the second string: "Saturday"
Output: 3  
*/
```

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@nutanaarohi123 add time and space complexity at the end of the file.

**Comment by iamrajiv** [comment]
@nutanaarohi123 remove `InterpolationSearch.java` file it's not related to this PR.

**Comment by nutanaarohi123** [comment]
Updated EditDistanceDP.c file and made changes as required. Removed the Interpolation java file.
Kindly review @iamrajiv

File: C/README.md [hunk 3]
```diff
@@ -46,8 +46,7 @@ _add list here_
 - [Binary Exponentiation](math/Binary_Exponentiation.c)
 - [GCD By Euclidean Algorithm](math/GCD In C.c)
 ## Dynamic Programming
-
-_add list here_
+- [EditDistance in Dp](dynamicprogram/EditDistanceDP.c)
```
Comment (iamrajiv) [inline C/README.md]: @nutanaarohi123 wrong URL added. Update this to `- [EditDistance in Dp](dp/EditDistanceDP.c)`

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@nutanaarohi123 add time and space complexity

File: C/README.md (hunk 3 above)
Comment (nutanaarohi123) [reply C/README.md]: Updated README.md and added Time and space complexity in the description section. of the pull requests.
kindly review @iamrajiv

File: C/dp/EditDistanceDP.c [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                       /* If last characters are same, ignore last char 
+			   and recur for remaining string */
+			else if (str1[indexstr1 - 1] == str2[indexstr2 - 1]) 
+				dp[indexstr1][indexstr2] = dp[indexstr1 - 1][indexstr2 - 1]; 
+                      /* If the last character is different, consider all 
+			  possibilities and find the minimum */
+			else
+                                dp[indexstr1][indexstr2] = 1 + mincount(dp[indexstr1][indexstr2 - 1], // Insert 
+							        dp[indexstr1 - 1][indexstr2], // Remove 
+								dp[indexstr1 - 1][indexstr2 - 1]); // Replace
```
Comment (iamrajiv) [inline C/dp/EditDistanceDP.c]: @nutanaarohi123 format the whole code using code formatter there is lot whitespaces and indentations

**Review by iamrajiv — CHANGES_REQUESTED** [review]

File: C/dp/EditDistanceDP.c (hunk 4 above)
Comment (nutanaarohi123) [reply C/dp/EditDistanceDP.c]: Updated the code and made changes as required. Kindly review @iamrajiv

**Review by Kajol-Kumari — APPROVED** [review]
Functionality working Fine! Well formatted and it's good to go :)

**Review by SKAUL05 — APPROVED** [review]
Well documented @nutanaarohi123 . Good to go :)

**Review by iamrajiv — APPROVED** [review]

**Comment by iamrajiv** [comment]
@all-contributors please add @nutanaarohi123 for code

**Pull request merged**